### PR TITLE
[gen-manifest] Initial impl of CEM generator

### LINKED ISSUE #2997: [labs/gen-manifest] Add initial implementation of CEM generator

### PR DESCRIPTION
**Pull request by kevinpschaaf**
Fixes #2997

Copious TODOs logged in https://github.com/lit/lit/issues/2993.

### CODE REVIEW

File: packages/labs/gen-manifest/goldens/test-element-a/custom-elements.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  "schemaVersion": "1.0.0",
+  "modules": [
+    {
+      "kind": "javascript-module",
+      "path": "element-a.js",
+      "summary": "TODO",
+      "description": "TODO",
+      "declarations": [
+        {
+          "kind": "variable",
```
Comment (justinfagnani) [inline packages/labs/gen-manifest/goldens/test-element-a/custom-elements.json:11]: Can you add a variable that's of type `ElementA` to see if the type reference works?

File: packages/labs/gen-manifest/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+  return cemReferences;
+};
+
+const convertReference = (reference: Reference): cem.TypeReference => {
+  return {
+    name: reference.name,
+    ...(reference.package
+      ? {package: reference.isGlobal ? 'global:' : reference.package}
+      : {}),
+    ...(reference.name ? {name: reference.name} : {}),
```
Comment (justinfagnani) [inline packages/labs/gen-manifest/src/index.ts:167]: `.name` should be `.module`? Add test

**Review by justinfagnani — APPROVED** [review]

File: packages/labs/gen-manifest/src/index.ts (hunk 2 above)
Comment (kevinpschaaf) [reply packages/labs/gen-manifest/src/index.ts]: Done

File: packages/labs/gen-manifest/package.json [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      ],
+      "#comment": "The quotes around the file regex must be double quotes on windows!",
+      "command": "uvu test \"_test\\.js$\"",
+      "output": []
+    }
+  },
+  "dependencies": {
+    "@lit-labs/analyzer": "^0.3.0",
+    "@lit-labs/gen-utils": "^0.1.0",
+    "custom-elements-manifest": "webcomponents/custom-elements-manifest#f893b20"
```
[the remark this answers is not in the document]
Comment (kevinpschaaf) [reply packages/labs/gen-manifest/package.json]: Done

File: packages/labs/gen-manifest/goldens/test-element-a/custom-elements.json (hunk 1 above)
Comment (kevinpschaaf) [reply packages/labs/gen-manifest/goldens/test-element-a/custom-elements.json]: Done. Added a few more to test the various types of references.

**Pull request merged**